examples/controlnet/train_controlnet.py

```diff
@@ -0,0 +1,818 @@
+            repo = Repository(args.output_dir, clone_from=repo_name, token=args.hub_token)
+
+            with open(os.path.join(args.output_dir, ".gitignore"), "w+") as gitignore:
+                if "step_*" not in gitignore:
+                    gitignore.write("step_*\n")
+                if "epoch_*" not in gitignore:
+                    gitignore.write("epoch_*\n")
+        elif args.output_dir is not None:
+            os.makedirs(args.output_dir, exist_ok=True)
+
+    # Load the tokenizer
+    if args.tokenizer_name:
+        tokenizer = AutoTokenizer.from_pretrained(args.tokenizer_name, revision=args.revision, use_fast=False)
+    elif args.pretrained_model_name_or_path:
+        tokenizer = AutoTokenizer.from_pretrained(
+            args.pretrained_model_name_or_path,
+            subfolder="tokenizer",
+            revision=args.revision,
+            use_fast=False,
+        )
+
+    # import correct text encoder class
+    text_encoder_cls = import_model_class_from_model_name_or_path(args.pretrained_model_name_or_path, args.revision)
+
+    # Load scheduler and models
+    noise_scheduler = DDPMScheduler.from_pretrained(args.pretrained_model_name_or_path, subfolder="scheduler")
+    text_encoder = text_encoder_cls.from_pretrained(
+        args.pretrained_model_name_or_path, subfolder="text_encoder", revision=args.revision
+    )
+    vae = AutoencoderKL.from_pretrained(args.pretrained_model_name_or_path, subfolder="vae", revision=args.revision)
+    unet = UNet2DConditionModel.from_pretrained(
+        args.pretrained_model_name_or_path, subfolder="unet", revision=args.revision
+    )
+
+    if args.controlnet_from_model:
+        print("Loading controlnet weights from model")
+        controlnet = ControlNetModel.from_pretrained(
+            args.pretrained_model_name_or_path, subfolder="controlnet", revision=args.revision
+        )
+    else:
+        print("Copying unet weight to controlnet")
+        # Create controlnet from unet weights.
+        config = dict(unet.config)
+
+        config["_class_name"] = "ControlNetModel"
+        config.pop("conv_out_kernel", None)
+        config.pop("out_channels", None)
+        config.pop("up_block_types", None)
+        config.pop("center_input_sample", None)
+        config.pop("conv_in_kernel", None)
+        config.pop("dual_cross_attention", None)
+        config.pop("mid_block_type", None)
+        config.pop("sample_size", None)
+        config.pop("time_cond_proj_dim", None)
+        config.pop("time_embedding_type", None)
+        config.pop("timestep_post_act", None)
+
+        controlnet = ControlNetModel.from_config(config)
+
+        unet_dict = unet.state_dict()
+        temp_state = controlnet.state_dict()
+
+        for k in temp_state.keys():
+            if k in unet_dict:
+                temp_state[k] = unet_dict[k].clone()
+            else:
+                if "controlnet" not in k:
+                    print("Not found in unet:", k)
```

Comment:
Instead of removing the unnecessary arguments from the unet, we should pass only the needed arguments to the controlnet. I think this also is a bit cleaner as a class method on the controlnet class